docs/apm/service-maps.asciidoc

```diff
@@ -0,0 +1,50 @@
+[[service-maps]]
+=== Service maps
+
+beta::[]
+
+A service map is a real-time diagram of the interactions occurring in your application’s architecture.
+It allows you to easily visualize data flow, and high-level statistics, like average transaction duration,
+requests per minute, errors per minute, and metrics, allow you to quickly assess the status of services.
+
+Our beta offering creates two types of service maps:
+
+* Global: All services and connections are shown.
+* Localized: Selecting a specific service will highlight it's connections.
+
+[role="screenshot"]
+image::apm/images/service-maps.png[Example view of service maps in the APM app in Kibana]
+
+[float]
+[[visualize-your-architecture]]
+=== Visualize your architecture
+
+Select the **Service Map** tab to get started.
+By default, all services and connections are shown.
+Whether your onboarding a new engineer, or just trying to grasp the big picture,
+click around, zoom in and out, and begin to visualize how your services are connected.
+
+If there's a specific service that interests you, select that service to highlight its connections.
+Clicking **Focus map** will refocus the map on that specific service and lock the connection highlighting.
+From here, select **Service Details**, or click on the **Transaction** tab to jump to the Transaction overview.
+You can also use the tabs at the top of the page to easily jump to the **Errors** or **Metrics** overview.
+
+While it's not possible to query in service maps, it is possible to filter by environment.
+This can be useful if you have two or more services, in separate environments, but with the same name.
+Use the environment drop down to only see the data you're interested in, like `dev` or `production`.
+
+[role="screenshot"]
+image::apm/images/service-maps-java.png[Example view of service maps with Java highlighted in the APM app in Kibana]
+
+[float]
+[[service-maps-legend]]
+=== Legend
+
+There are four type of icons.
```

Comment:
Not sure if the breakdown here is clear or helpful. There's two shapes: circle for instrumented services and diamond for databases/external/messaging. Each of the shapes have icons inside. In the case of services it shows the language agent used, and for the diamond items it uses a generic icon representing the type of thing or more specific icon (currently the only thing in a diamond that has a specific icon is Elasticsearch, but we're planning on adding icons for things like postgres or redis or kafka.)

Reply:
Thanks. I updated it. I agree though, I'm not sure how useful this is. Might be worth just removing entirely.